frame/contracts/src/lib.rs

```diff
@@ -575,21 +574,16 @@ decl_module! {
 		/// You can instantiate contracts only with stored code.
 		pub fn put_code(
```

Comment:
>  although yeah with a rather pricy transaction byte fee. I will change the price down the line.

You could start experimenting with `FunctionOf()` here and make this guys weight a function of the code length, if that is what you are looking for here.

Reply:
Yeah, I think it would be actually great!